CMakeLists.txt

```
@@ -8,8 +8,25 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/$<CONFIGURATION>)
 # Declare the project
 project(sdl-min)
 
-# Create an executable and add our sources to it.
-add_executable(${PROJECT_NAME} src/main.cpp src/iosLaunchScreen.storyboard src/Sample.appxManifest)
+# Add all the sources
+# set(SRC_PATH src)
+# set(SOURCES ${SRC_PATH}/main.cpp)
+
+# Create an executable or a shared library based on the platform and add our sources to it
+if (!ANDROID)
+	add_executable(${PROJECT_NAME})
+else()
+	add_library(${PROJECT_NAME} SHARED)
+	target_include_directories(${PROJECT_NAME} PRIVATE SDL/include/SDL3 SDL/include)
+endif()
```

This is not doing what you think, `!ANDROID` is treated as a variable name.

Instead you probably want:
```suggestion
if (ANDROID)
	add_library(${PROJECT_NAME} SHARED)
else()
	add_executable(${PROJECT_NAME})
endif()
```

Also, the `target_include_directories` is redundant since you link `SDL3::SDL3`.